lib/std/fs/test.zig

```diff
@@ -907,6 +907,25 @@ test "makePath in a directory that no longer exists" {
     try testing.expectError(error.FileNotFound, tmp.dir.makePath("sub-path"));
 }
 
+test "makePath but sub_path contains pre-existing file" {
+    // makePath tmp/foo/bar/baz, but bar is a file
+    var buf: [std.fs.MAX_PATH_BYTES]u8 = undefined;
+    var fba = std.heap.FixedBufferAllocator.init(&buf);
+    var buf_alloc = fba.allocator();
+
+    var tmp = tmpDir(.{});
+    defer tmp.cleanup();
+    
+    try tmp.dir.makeDir("foo");
+    var foo = try tmp.dir.openDir("foo", .{});
+    defer foo.close();
+    var bar = try foo.createFile("bar", .{});
+    defer bar.close();
+    
+    var sub_path = try std.fs.path.join(buf_alloc, &[_][]const u8{"foo", "bar", "baz"});
+    try testing.expectError(error.NotDir, tmp.dir.makePath(sub_path));
+}
```

Comment:
Just to simplify things a bit:

```suggestion
test "makePath but sub_path component is a pre-existing file" {
    var tmp = tmpDir(.{});
    defer tmp.cleanup();

    try tmp.dir.makeDir("foo");
    try tmp.dir.writeFile("foo/bar", "");
    
    try testing.expectError(error.NotDir, tmp.dir.makePath("foo/bar/baz"));
}
```

Reply:
I like the simplification, but is this guaranteed to work in all instances regardless of what the native path separator is?

Reply:
The only target it would fail on is UEFI (which only supports `\` AFAIK), but there are other tests in `fs/test.zig` that use `/` as the path separator so it should be fine for now:

https://github.com/ziglang/zig/blob/ddf5859c22527c6bf5d8bb13310db996fcc58874/lib/std/fs/test.zig#L454

Reply:
If you want to use the native path separator, though, this would also work:

```zig
test "makePath but sub_path component is a pre-existing file" {
    var tmp = tmpDir(.{});
    defer tmp.cleanup();
    try tmp.dir.makeDir("foo");
    try tmp.dir.writeFile("foo" ++ fs.path.sep_str ++ "bar", "");
    
    try testing.expectError(error.NotDir, tmp.dir.makePath("foo" ++ fs.path.sep_str ++ "bar" ++ fs.path.sep_str ++ "baz"));
}